pkg/apis/vpc/v1alpha1/subnetipreservation_types.go

```diff
@@ -0,0 +1,64 @@
+/* Copyright © 2025 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetIPReservationSpec defines the desired state of SubnetIPReservation
+type SubnetIPReservationSpec struct {
+	// Subnet specifies the Subnet to reserve IPs from.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Subnet is immutable"
+	Subnet string `json:"subnet"`
+
+	// RequestedNumberOfIPs defines number of IPs requested to be reserved.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="RequestedNumberOfIPs is immutable"
+	// +kubebuilder:validation:Maximum:=100
+	// +kubebuilder:validation:Minimum:=1
+	RequestedNumberOfIPs int `json:"requestedNumberOfIPs"`
+}
+
+// SubnetIPReservationStatus defines the observed state of SubnetIPReservation
+type SubnetIPReservationStatus struct {
+	// Conditions described if the SubnetIPReservation is configured on NSX or not.
+	// Condition type ""
+	Conditions []Condition `json:"conditions,omitempty"`
+	// List of reserved IPs.
+	// Supported formats include: ["192.168.1.1", "192.168.1.3-192.168.1.100"]
+	IPs []string `json:"ips,omitempty"`
+}
+
+//+genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//kubebuilder:storageversion
+
+// SubnetIPReservation is the Schema for the subnetipreservations API
+// +kubebuilder:printcolumn:name="Subnet",type=string,JSONPath=`.spec.subnet`,description="The parent Subnet name of the SubnetIPReservation."
+// +kubebuilder:printcolumn:name="RequestedNumberOfIPs",type=string,JSONPath=`.spec.requestedNumberOfIPs`,description="Number of IPs requested to be reserved."
+// +kubebuilder:printcolumn:name="IPs",type=string,JSONPath=`.status.ips[:]`,description="List of reserved IPs."
+type SubnetIPReservation struct {
```

Comment:
To confirm - in NSX the API is called `IPReservation` too?

Reply:
According to their design doc, they will support
` /orgs/<org-id>/projects/<project-id>/vpcs/<vpc-id>/subnets/<subnet-id>/dynamic-ip-reservations/<id>`
for allocating N IPs
And use
`/orgs/<org-id>/projects/<project-id>/vpcs/<vpc-id>/subnets/<subnet-id>/static-ip-reservations/<id>`
for allocating specific IP which will be supported after 9.1
But their API patch has not merged yet.